src/go/rpk/pkg/cli/cmd/wasm/deploy.go

```diff
@@ -96,38 +97,40 @@ func deploy(
 		}
 	}
 	// create headers
-	headers, err := createHeaders("deploy", description, fileContent)
+	headers, err := createHeaders("deploy", description, fileContent, fileName)
 	if err != nil {
 		return err
 	}
+	// create id from filename
+	id := xxhash.Sum64([]byte(fileName))
 	// publish message
-	error := kafka.PublishMessage(producer, fileContent, fileName, kafka.CoprocessorTopic, headers)
-	if error != nil {
-		return fmt.Errorf("error deploying '%s.js: %v'", fileName, error)
+	err = kafka.PublishMessage(
+		producer,
+		fileContent,
+		strconv.FormatUint(id, 10),
+		kafka.CoprocessorTopic,
+		headers,
+	)
+	if err != nil {
+		return fmt.Errorf("error deploying '%s.js: %v'", fileName, err)
 	}
 	return nil
 }
 
 func createHeaders(
-	action string, description string, content []byte,
+	action string, description string, content []byte, filename string,
 ) ([]sarama.RecordHeader, error) {
-	var headersResult []sarama.RecordHeader
-	// create simple struct for key string and value string
-	value := []struct {
-		key	string
-		value	string
-	}{
-		{key: "action", value: action},
-		{key: "description", value: description},
-	}
-	// create RecordHeader and append to headersResult
-	for _, v := range value {
-		keyH := []byte(v.key)
-		valueH := []byte(v.value)
-		headersResult = append(headersResult, struct {
-			Key	[]byte
-			Value	[]byte
-		}{Key: keyH, Value: valueH})
+	var headersResult []sarama.RecordHeader = []sarama.RecordHeader{
+		{
+			Key:	[]byte("action"),
+			Value:	[]byte(action),
+		}, {
+			Key:	[]byte("description"),
+			Value:	[]byte(description),
+		}, {
+			Key:	[]byte("file_name"),
```

Comment:
Can we change the name of the key to `id` instead of `filename`? It is a bytes type which isn't human readable anyway.

Reply:
When I read `coprocessor_internal_topic` I can read the `file_name`: 

```{
 "headers": [
  {
   "key": "action",
   "value": "remove"
  },
  {
   "key": "file_name",
   "value": "elastic.js"
  }
 ],
```

but, I like `id` as key, I'll do it

Reply:
After some discussion there was some confusion on my part. The header containing the filename, should have a key named 'file_name'. 'id' shouldn't be there as the true id (bytes type) is the key of the record itself.